internal/collector/otel_collector_plugin.go

```diff
@@ -117,19 +119,182 @@ func (oc *Collector) Close(ctx context.Context) error {
 	slog.InfoContext(ctx, "Closing OTel Collector plugin")
 
 	if !oc.stopped {
+		slog.InfoContext(ctx, "Shutting down OTel Collector", "state", oc.service.GetState())
 		oc.stopped = true
 		oc.service.Shutdown()
+		oc.cancel()
+
+		settings := oc.config.Common
+		settings.MaxElapsedTime = maxTimeToWaitForShutdown
+		err := backoff.WaitUntil(ctx, oc.config.Common, func() error {
+			if oc.service.GetState() == otelcol.StateClosed {
+				return nil
+			}
+
+			return errors.New("OTel Collector not in a closed state yet")
+		})
+
+		if err != nil {
+			slog.ErrorContext(ctx, "Failed to shutdown OTel Collector", "error", err, "state", oc.service.GetState())
+		} else {
+			slog.InfoContext(ctx, "OTel Collector shutdown", "state", oc.service.GetState())
+		}
 	}
-	<-oc.appDone
 
 	return nil
 }
 
 // Process an incoming Message Bus message in the plugin
-func (oc *Collector) Process(_ context.Context, msg *bus.Message) {
+func (oc *Collector) Process(ctx context.Context, msg *bus.Message) {
+	switch msg.Topic {
+	case bus.NginxConfigUpdateTopic:
+		oc.handleNginxConfigUpdate(ctx, msg)
+	default:
+		slog.DebugContext(ctx, "OTel collector plugin unknown topic", "topic", msg.Topic)
+	}
 }
 
 // Subscriptions returns the list of topics the plugin is subscribed to
 func (oc *Collector) Subscriptions() []string {
-	return []string{}
+	return []string{
+		bus.NginxConfigUpdateTopic,
+	}
+}
+
+func (oc *Collector) handleNginxConfigUpdate(ctx context.Context, msg *bus.Message) {
+	nginxConfigContext, ok := msg.Data.(*model.NginxConfigContext)
+	if !ok {
+		slog.ErrorContext(ctx, "Unable to cast message payload to *model.NginxConfigContext", "payload", msg.Data)
```

Comment:
Probably need a `return` here too.
```suggestion
		slog.ErrorContext(ctx, "Unable to cast message payload to *model.NginxConfigContext", "payload", msg.Data)
```
```suggestion
		slog.ErrorContext(ctx, "Unable to cast message payload to *model.NginxConfigContext", "payload", msg.Data)
		
		return
```